### ISSUE TITLE
[HOLD for payment 2021-09-16] Desktop - Big empty space in message fields if copy/paste the Text from MAC Note.app

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#4484]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
Issue was found when executing the PR #4363
## Action Performed:
1. Navigate to a conversation that has some messages.
2. Paste the Text from MAC note.app to the composer. 
## Expected Result:
Only entered message should be visible in compose field if copy/paste the Text from MAC Note.app

## Actual Result:
Big empty space in message compose field if copy/past the Text from MAC Note.app

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Desktop App

**Version Number:**
v1.0.83-0
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/128581157-fbeefcae-5559-4e17-b406-ed462d15c44c.mp4

![Bug5182815_empty](https://user-images.githubusercontent.com/43996225/128581159-d95cb257-877c-414e-bc38-bcf1e57754b1.jpg)

**Expensify/Expensify Issue URL:**

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @bondydaa (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by bondydaa** [comment]
I am OOO

**Comment by francoisl** [comment]
@parasharrajat - just checking, it looks like you were [aware](https://github.com/Expensify/App/pull/4363#issuecomment-890960388) of this issue, correct? Was there a separate GH issue that was opened to track the bug where newlines are prepended to messages pasted into the message box?

**Comment by parasharrajat** [comment]
No, a new issue was opened at the moment. I was waiting for  #4363 to merge as only after that, this behaviour is revealed. 
So with an inspection, I come to know that the Mac Note app. usages `&nbsp;` chars to show spaces which are also needed to be escaped and current `_.unescape` does not do that. I think we need a custom implementation as is not made to parse HTMLEntities and its usages do not seem sufficient any more.

About the main issue, let me quickly take a good peek at it. Only then I can tell what is causing this much space.

**Comment by roryabraham** [comment]
I think we should demote this from deploy blocker to regular-old 🐛

**Comment by marcaaron** [comment]
This hasn't been looked at in 9 days and has no assignee. Bumping to daily and making it External so we do something.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @kadiealexander (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by kadiealexander** [comment]
@parasharrajat is this related at all to https://github.com/Expensify/App/issues/4759?

**Comment by parasharrajat** [comment]
@kadiealexander https://github.com/Expensify/App/issues/4759 is needed to improve the parsing. But I was not able to reproduce the same behavior of this issue. There is space between pasted text but not before or after.

**Comment by parasharrajat** [comment]
After fixing https://github.com/Expensify/App/issues/4759, 
I get this where I see no space.

https://user-images.githubusercontent.com/24370807/130372692-796af653-411b-43ac-b749-6c55c75678dc.mp4

**Comment by kadiealexander** [comment]
What version are you working with @parasharrajat? I just tested on version 1.0.85-9 (1.0.85-9) and I still see the white space above the message:

https://user-images.githubusercontent.com/59587260/130373230-a3d23ffe-d83f-41ea-86d5-4336f58c38a3.mp4

**Comment by parasharrajat** [comment]
Ok. I think it's specific to chrome. Let me retest.

**Comment by kadiealexander** [comment]
Mine was on desktop app too, just FYI for the thread. 

Thanks for helping me troubleshoot it! Will post to UW now :)

**Comment by parasharrajat** [comment]
I have found the fix as well. Preparing the proposal now.

**Comment by kadiealexander** [comment]
Upwork job: https://www.upwork.com/jobs/~01078e92ead30d76a6

**Comment by MelvinBot** [comment]
Triggered auto assignment to @johnmlee101 (`Exported`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by kadiealexander** [comment]
@johnmlee101 when you get a moment, would you mind reviewing the existing proposals here?

**Comment by johnmlee101** [comment]
Love the proposal @parasharrajat .

@kadiealexander feel free to assign out the job!

**Comment by johnmlee101** [comment]
Bumping back to weekly while we wait for this to close out

**Comment by parasharrajat** [comment]
I am on vacation. back on 8.

**Comment by botify** [comment]
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

**Comment by botify** [comment]
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/1214907189) to staging by @johnmlee101 in version: 1.0.94-2 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

**Comment by botify** [comment]
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/1218876549) to production by @yuwenmemon in version: 1.0.95-1 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

**Comment by parasharrajat** [comment]
@kadiealexander Any update for Upwork?

**Comment by kadiealexander** [comment]
Sorry @parasharrajat, timezones meant this fell on my weekend! I've paid this now. 💸
